### PR TITLE
Fix spatial mapping to work with teleport and camera control.

### PR DESCRIPTION
**Pull request by fast-slow-still**
## Overview
Modification to allow spatial mapping to work with teleport and camera control via the MixedRealityPlayspace.

Rather than having a WorldAnchor directly control the spatial mesh's transform, read the WorldAnchor's transform and concatenate with the playspace transform to get the mesh's transform.

## Changes
- Fixes: #4276  .

> As a reviewer, it is possible to check out this change locally by using the following
> commands (substituting {PR_ID} with the ID of this pull request):
>
> git fetch origin pull/{PR_ID}/head:name_of_local_branch
>
> git checkout name_of_local_branch

### CODE REVIEW

**Comment by julenka** [comment]
From the bug:

> Add 1 meter offset on Y in MixedRealityPlayspace position

I thought that you are not supposed to move the playspace, according to @simondarkside. Am I misunderstanding something?

**Comment by SimonDarksideJ** [comment]
This is a long running debate @julenka 
* Immersive HAS to start at 0 as the height is determined by the framework
* AR requires the head position to be relative to the players head, manual.

@Railboy what are your thoughts, did we ever get the head setup thing sorted?

**Comment by fast-slow-still** [comment]
As far as whether it’s okay to change the transform on the playspace, I believe that is the reason we have the playspace. Note that teleport is implemented by adjusting the transform of the playspace. Manual camera control is generally enabled by putting a ManualCameraControl component on the playspace. (Putting the camera control component on the camera would be ineffective since the camera’s transform is overwritten each frame to track the head).

I think the misconception here stems from thinking of a transform on the MRTK Playspace as moving the camera. In AR, that’s obviously not what we want. The camera’s transform is already keeping the camera aligned with the head, where it should be.

It might help to think of the MRTK Playspace transform as repositioning everything in the scene relative to the head. Moreover, it repositions the contents of the scene without changing their global scene coordinates, so it won’t break any physics simulation going on.

This is a very powerful and commonly used mechanism that seems important to support. 

@julenka , @SimonDarksideJ , It seems like there's an ongoing conversation I haven't seen. I'd be interested in understanding your thoughts better. Is there a link? Or could you summarize? I'm wondering why we wouldn't want to support this, especially since we clearly do need to support it in immersive.

File: Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealitySpatialMeshObserver.cs [hunk 1]
```diff
@@ -422,7 +452,18 @@ private void RequestMesh(SurfaceId surfaceId)
...
-                worldAnchor = newMesh.GameObject.AddComponent<WorldAnchor>();
+                // The WorldAnchor component places its object where the anchor is in the same space as the camera. 
+                // But since the camera is repositioned by the MixedRealityPlayspace's transform, the meshes' transforms
+                // should also the WorldAnchor position repositioned by the MixedRealityPlayspace's transform.
+                // So rather than put the WorldAnchor on the mesh's GameObject, the WorldAnchor is placed out of the way in the scene,
+                // and its transform is concatenated with the Playspace transform to compute the transform on the mesh's object.
+                // That adapting the WorldAnchor's transform into playspace is done by the internal PlayspaceAdapter component.
+                // The GameObject the WorldAnchor is placed on is unimportant, but it is convenient for cleanup to make it a child
+                // of the GameObject whose transform will track it.
+                GameObject anchorHolder = new GameObject(meshName + "_anchor");
```
Comment (StephenHodgson) [inline Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealitySpatialMeshObserver.cs]: Do we ever clean up this GameObject when the spatial object is recycled?

Comment (fast-slow-still) [reply Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealitySpatialMeshObserver.cs]: The anchorHolder is attached as a child node to the newMesh.GameObject. It is cleaned up as part of the newMesh.GameObject tree.
